builtin/mainmenu/dlg_config_world.lua

```diff
@@ -160,17 +171,29 @@ local function handle_buttons(this, fields)
 		local rawlist = this.data.list:get_raw_list()
 
 		local i,mod
+		-- mod_decisions helps keeping track of what value to assign to the
+		-- load_mod_modname setting, as only one mod with a particular name
+		-- can be set.
+		-- Values: nil = not yet decided, otherwise the value of mod.enabled
+		-- for the mod that was chosen.
+		local mod_decisions = {}
 		for i,mod in ipairs(rawlist) do
+			local load_mod_setting = "load_mod_"..mod.name
 			if not mod.is_modpack and
 					not mod.is_game_content then
 				if modname_valid(mod.name) then
-					worldfile:set("load_mod_"..mod.name, tostring(mod.enabled))
-				else
-					if mod.enabled then
-						gamedata.errormessage = fgettext_ne("Failed to enable mod \"$1\" as it contains disallowed characters. Only chararacters [a-z0-9_] are allowed.", mod.name)
+					local decision = mod_decisions[mod.name]
+					-- Is it either enabled or the first one seen?
+					if decision == nil or mod.enabled then
+						mod_decisions[mod.name] = mod.enabled
+						worldfile:set(load_mod_setting, mod.enabled and get_modpack_path(mod) or "false")
```

Comment:
Line too long (> 80 characters)

Reply:
Following the Lua style guide literally, only the word "false" exceeds the 80th character, which could be interpreted as being OK. Where should I split?